crates/oxc_linter/src/rules/eslint/no_undef.rs

```diff
@@ -31,9 +31,38 @@ declare_oxc_lint!(
     /// var bar = a + 1;
     /// ```
     NoUndef,
-    nursery
+    correctness,
+    pending
 );
 
+const TS_UTILITY_TYPES: [&str; 25] = [
```

Comment:
While I don't actually know how to implement it, @Boshen mentioned checking for symbol flags in the issue.  I assume that would mean not using a hard coded list like this.

Reply:
hmm yeah this is rough, we would have to keep this list exactly up to date with the globals TS requires.

also, if a user was using a package that declares a global (type or var) this would break

Reply:
I didn't found the ts types list in the repo. So this a first proposal to raise this question too :)